components/RunOverview.vue

```diff
@@ -45,6 +48,22 @@ export default class RunOverview extends Vue {
     return this.run.Metadata.Variables?.Variable;
   }
 
+  @Watch('run.GameName', {immediate: true})
+  coverSource() {
+    const url = `https://www.speedrun.com/api/v1/games?name=${encodeURIComponent(this.run.GameName)}`;
+
+    fetch(url)
```

Comment:
Also, thanks to how awesome promises are, the "stop going further part" is quite easy : if you detect an error in the first `then`, you can throw an exception, which will make the second `then` not execute, and then catch said exception in a `.catch` at the end of the promise chain. Which means that when you detect an error case you can just `throw 'my error message or a value or whatever';` and then handle all errors in a single `.catch` at then end (since we want to do the same thing every time anyway)